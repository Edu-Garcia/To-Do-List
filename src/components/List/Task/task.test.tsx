import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import { Task } from '.'
import '@testing-library/jest-dom';

describe('Task Test', () => {
  it('snapshot test task uncompleted', () => {
    const mockTask = {
      id: '1',
      title: 'Tarefa 1',
      description: 'Descrição da tarefa 1',
      complete: false,
      user: null
    }

    const { container } = render(
      <BrowserRouter>
        <Task
          completeTask={jest.fn()}
          deleteTask={jest.fn()}
          editTask={jest.fn()}
          key={1}
          {...mockTask}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('snapshot test task completed', () => {
    const mockTask = {
      id: '1',
      title: 'Tarefa 1',
      description: 'Descrição da tarefa 1',
      complete: true,
      user: null
    }

    const { container } = render(
      <BrowserRouter>
        <Task
          completeTask={jest.fn()}
          deleteTask={jest.fn()}
          editTask={jest.fn()}
          key={1}
          {...mockTask}
        />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});