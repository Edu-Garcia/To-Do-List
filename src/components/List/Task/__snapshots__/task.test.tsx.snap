// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Test snapshot test task completed 1`] = `
<div>
  <li
    class="selectedTask"
  >
    <div
      class="textContainer"
    >
      <h3>
        Tarefa 1
      </h3>
      <span>
        Descrição da tarefa 1
      </span>
    </div>
    <button
      class="button edit"
      type="button"
    >
      Editar
    </button>
    <button
      class="button delete"
      type="button"
    >
      Excluir
    </button>
  </li>
</div>
`;

exports[`Task Test snapshot test task uncompleted 1`] = `
<div>
  <li
    class=""
  >
    <div
      class="textContainer"
    >
      <h3>
        Tarefa 1
      </h3>
      <span>
        Descrição da tarefa 1
      </span>
    </div>
    <button
      class="button undefined"
      type="button"
    >
      Concluir
    </button>
    <button
      class="button edit"
      type="button"
    >
      Editar
    </button>
    <button
      class="button delete"
      type="button"
    >
      Excluir
    </button>
  </li>
</div>
`;
